compiler-core/src/parse/extra.rs

```diff
@@ -7,7 +7,9 @@ use crate::ast::SrcSpan;
 #[derive(Debug, PartialEq, Eq, Default)]
 pub struct ModuleExtra {
     pub module_comments: Vec<SrcSpan>,
-    pub doc_comments: Vec<SrcSpan>,
+    /// Contains location information for each documentation comment, including the position
+    /// right before its '///' marker starts, as well as the span of its contents (after the marker).
+    pub doc_comments: Vec<(u32, SrcSpan)>,
```

Comment:
Isn't this u32 always the same as the `start` of the SrcSpan?

Reply:
The `start` of the SrcSpan points to the `x` in `///x`, whereas the u32 points to the first `/` in `///x`. I just wanted to distinguish the marker start from the content start, basically, as comments always use the content spans as their own spans, not considering their markers.

I guess we could switch to just taking the content start and subtracting 3 instead of storing the marker start if a doc comment can never exist without `///` anyway.

Reply:
It doesn't matter which we go for, but we shouldn't duplicate the data here. One is always the other + 3, so it can be calculated when needed.